src/main/kotlin/lotto/domain/LottoNumber.kt

```diff
@@ -0,0 +1,19 @@
+package lotto.domain
+
+class LottoNumber private constructor(
+    val value: Int,
+) : Comparable<LottoNumber> {
+    override fun compareTo(other: LottoNumber): Int {
+        return value.compareTo(other.value)
+    }
+
+    companion object {
+        private const val MINIMUM_LOTTO_NUMBER = 1
+        private const val MAXIMUM_LOTTO_NUMBER = 45
+
+        private val cache = (MINIMUM_LOTTO_NUMBER..MAXIMUM_LOTTO_NUMBER).associateWith { LottoNumber(it) }
+        val cached = cache.keys
```

Comment:
Nitpicking: 
How about specifying each explicit type like below? This makes it easier to inspect. 🙂 
```suggestion
        private val cache: Map<Int, LottoNumber> = (MINIMUM_LOTTO_NUMBER..MAXIMUM_LOTTO_NUMBER).associateWith { LottoNumber(it) }
        val cached: Set<Int> = cache.keys
```

Reply:
Reasonable 🙌
I like the feature of Kotlin's type inference, but sometimes it is hard to expect what the type is.
I tend to get confused when to specify or not. 
I know that intellij provides feature for `type inlays` but I just wanted to know if you have any specific circumstances where you specify types?